superset-frontend/src/features/home/ActivityTable.test.tsx

```diff
@@ -85,55 +78,61 @@ describe('ActivityTable', () => {
     isFetchingActivityData: false,
   };
 
-  let wrapper: ReactWrapper;
+  const activityViewedTabProps = {
+    activeChild: TableTab.Viewed,
+    activityData: mockData,
+    setActiveChild: jest.fn(),
+    user: { userId: '1' },
+    isFetchingActivityData: false,
+  };
 
-  beforeAll(async () => {
-    await act(async () => {
-      wrapper = mount(
-        <Provider store={store}>
-          <ActivityTable {...activityProps} />
-        </Provider>,
-      );
-    });
-  });
+  const emptyActivityProps = {
+    activeChild: TableTab.Created,
+    activityData: {},
+    setActiveChild: jest.fn(),
+    user: { userId: '1' },
+    isFetchingActivityData: false,
+  };
 
-  it('the component renders', () => {
-    expect(wrapper.find(ActivityTable)).toExist();
+  const renderOptions = {
+    useRedux: true,
+    useRouter: true,
+  };
+
+  const renderActivityTable = (props: any) =>
+    render(<ActivityTable {...props} />, renderOptions);
+
+  it('the component renders with ActivityCards', async () => {
+    renderActivityTable(activityProps);
+    expect(screen.getByText(/dashboard_test/i)).toBeInTheDocument();
   });
-  it('renders tabs with three buttons', () => {
-    expect(wrapper.find('[role="tab"]')).toHaveLength(3);
+  it('renders tabs with three buttons', async () => {
+    renderActivityTable(activityProps);
+    expect(screen.getAllByRole('tab')).toHaveLength(3);
   });
-  it('renders ActivityCards', async () => {
-    expect(wrapper.find('ListViewCard')).toExist();
+  it('renders Viewed tab with ActivityCards', async () => {
+    renderActivityTable(activityViewedTabProps);
+    expect(screen.getByText(/chartychart/i)).toBeInTheDocument();
   });
   it('calls the getEdited batch call when edited tab is clicked', async () => {
-    act(() => {
-      const handler = wrapper.find('[role="tab"] a').at(1).prop('onClick');
-      if (handler) {
-        handler({} as any);
-      }
-    });
+    renderActivityTable(activityProps);
+    const editedButton = screen.getByText(/edited/i);
+    expect(editedButton).toBeInTheDocument();
+    userEvent.click(editedButton);
     const dashboardCall = fetchMock.calls(/dashboard\/\?q/);
     const chartCall = fetchMock.calls(/chart\/\?q/);
-    // waitforcomponenttopaint does not work here in this instance...
+    // waitFor() does not work here in this instance...
```

Comment:
could you try with
```
await waitFor(() => expect(fetchMock.calls(/chart\/\?q/).toHaveLength(1));
```

Reply:
that was actually my original attempt but the test will not pass, seems to be the same problem the previous author ran into. Not sure what the diagnosis is but it is passing with setTimeout

Reply:
attempted adding a 5s timeout to the `waitFor()` with no change in result but the test passes in <20ms when using `setTimeout`. the only feedback I'm getting is a deprecation warning from the `moment` library concerning use of `non RFC2822/ISO date formats` probably from the `changed_on_utc` value in the `fetchMock`s

Reply:
@rtexelm it appears that the API fetch is only triggered when the 
`activeChild` prop is updated to `TableTab.Edited`.
Hence, it's being called before the case, which seems to be incorrect.

https://github.com/apache/superset/blob/47ae9d4cc3ca94332cb14359fb8a306f91c2da60/superset-frontend/src/features/home/ActivityTable.tsx#L142-L146

Kindly consider using `rerender` with `activeChild={TableTab.Edited}` (which is anticipated to be propagated after `setActiveChild()` has been called). This method should be a right direction whether the fetch has been called correctly.

Reply:
Fantastic! I didn't know the usefulness of `rerender`. That looks like it's worked. The fetchMock calls don't register but the mockData is rendered without specifying `TableTab.Edited` as a replete tab